lib/IDE/SourceEntityWalker.cpp

```diff
@@ -422,6 +426,34 @@ std::pair<bool, Expr *> SemaAnnotator::walkToExprPre(Expr *E) {
     if (!walkToExprPost(E))
       return { false, nullptr };
     return { false, E };
+  } else if (auto OEE = dyn_cast<OpenExistentialExpr>(E)) {
+    // Record opaque value.
+    OpaqueValueMap[OEE->getOpaqueValue()] = OEE->getExistentialValue();
```

Comment:
How about removing the entries after it's walked the subexpression?

```
SWIFT_DEFER {
  OpaqueValueMap.erase(OEE->getOpaqueValue());
};
```
 
should do it.

Reply:
That's a good idea! Thanks!